### DISCUSSION TITLE
Automatic folder detection / indexing for new foilders

### DISCUSSION THREAD

**Discussion by update-freak** [photoprism/photoprism#2385] [Q&A]
Hi everyone,

is there an automatic folder detection / indexing for new folders? (instead of manually select the new folder and start the process of indexing)? I use Photoprism on Docker on Synology NAS.

**Comment by lastzero** [comment]
Start indexing without selecting a specific folder. Do not select the rescan option, so already indexed files and folders are skipped.

**Reply by update-freak** [reply]
But I have to do that every time I add new pictures or folders, right? I want an automatic mechanism that these steps are not necessary

**Comment by update-freak** [comment]
ok, thanks. Is there also a way that I did not have to start this manually when I add photos, so it is an automatic process?
I found out that here https://dl.photoprism.org/docker/scheduler/ is a scheduled solution mentioned.
I'm not sure if this will work with Docker on a Synology NAS.

**Reply by graciousgrey** [reply]
Unfortunately we don't have a Synology test device. Maybe a community member can help. 
Setting up background jobs in the web ui is planned: https://github.com/photoprism/photoprism/issues/2495

**Comment by update-freak** [comment]
Is it possible in Synology in Docker with?

docker run -d --name=Ofelia \
-v /var/run/docker.sock:ro:/var/run/docker.sock \
-v /volume1/docker/ofelia/config.ini:./jobs.ini \
--net=host \
--restart always \
mcuadros/ofelia


jobs.ini
[job-exec "photoprism index"]
schedule = 0 0 2 * * *
container = Photoprism
command =  photoprism index --cleanup
no-overlap = true

**Comment by update-freak** [comment]
Is it possible to use the command "sudo docker exec PhotoPrism /bin/sh /volume1/docker/photoprism index" to start idexing of new pictures?

I installed PhotoPrism in Docker with:
`docker run -d --name=PhotoPrism \
-p 2342:2342 \
-e PHOTOPRISM_UPLOAD_NSFW=true \
-e PUID=1026 \
-e PGID=100 \
-e TZ=Europe/Berlin \
-e PHOTOPRISM_ADMIN_PASSWORD=PASSWORD \
-v /volume1/docker/photoprism:/photoprism/storage \
-v /volume1/Bilder:/photoprism/originals \
--security-opt seccomp=unconfined \
--security-opt apparmor=unconfined \
photoprism/photoprism`

**Comment by update-freak** [comment]
found a solution with Synology schedule:

`sudo docker exec PhotoPrism /opt/photoprism/bin/photoprism index`
